contracts/ui/UiPoolDataProvider.sol

```diff
@@ -260,29 +261,60 @@ contract UiPoolDataProvider is IUiPoolDataProvider {
         return (reservesData, baseCurrencyInfo);
     }
 
+    function getAuctionData(
+        IPoolAddressesProvider provider,
+        address user,
+        address[] memory nTokenAddresses,
+        uint256[][] memory tokenIds
+    ) external view override returns (DataTypes.AuctionData[][] memory) {
+        uint256[] memory userBalances = new uint256[](nTokenAddresses.length);
+
+        uint256 tokenDataSize;
+
+        DataTypes.AuctionData[][]
+            memory tokenData = new DataTypes.AuctionData[][](
+                nTokenAddresses.length
+            );
+        IPool pool = IPool(provider.getPool());
+
+        for (uint256 i = 0; i < nTokenAddresses.length; i++) {
+            address asset = nTokenAddresses[i];
+            uint256 userTotalBalance = INToken(asset).balanceOf(user);
+            tokenData[i] = new DataTypes.AuctionData[](userTotalBalance);
+
+            for (uint256 j = 0; j < userTotalBalance; j++) {
```

Comment:
I think we can use `tokenIds[i].length` as the loop count instead of `userTotalBalance `, If we use `userTotalBalance`, userTotalBalance must be less than  `tokenIds[i].length`.